contentcuration/contentcuration/models.py

```diff
@@ -1478,12 +1501,26 @@ def has_changes(self):
         original_values = self._field_updates.changed()
         return any((True for field in original_values if field not in blacklist))
 
+    def recalculate_editors_storage(self):
+        from contentcuration.utils.user import calculate_user_storage
+        for editor in self.files.values_list('uploaded_by_id', flat=True):
+            calculate_user_storage(editor)
+
     def on_create(self):
         self.changed = True
+        self.recalculate_editors_storage()
 
     def on_update(self):
         self.changed = self.changed or self.has_changes()
 
+    def move_to(self, target, *args, **kwargs):
+        parent_was_trashtree = self.parent.channel_trash.exists()
+        super(ContentNode, self).move_to(target, *args, **kwargs)
+
+        # Recalculate storage if node was moved to or from the trash tree
+        if target.channel_trash.exists() or parent_was_trashtree:
+            self.recalculate_editors_storage()
```

Comment:
I think it may be simpler to remove this overridden move_to and the `on_create` line and add this into the `save` method below where we already have the `same_order` check logic. Can just add a final:

```
if not same_order:
    self.recalculate_editors_storage()
```
At the end of the `save` method.

Otherwise, I think there are some edge cases in copying that might escape this.

Reply:
If it's a copying method, would we still need to recalculate the storage?